OGM-716 FIx distirbution package

DavideD opened the pull request:
Hello can you guys have alook a the resulting distribution package and let me know?

It should be similar to the previous one, except for:
- There is on optional folder containing parboiled and hibernate search dependencies
- core contains hibernate-ogm-core and all the required dependencies

gunnarmorling commented on bom/pom.xml:
```diff
@@ -103,11 +105,12 @@
                 <groupId>org.hibernate</groupId>
                 <artifactId>hibernate-envers</artifactId>
                 <version>${hibernateVersion}</version>
-            </dependency>
```
Why did you remove this module from the BOM?

DavideD replied:
I removed it because we don't use it anywhere? Is there any particular reason to keep it?

Anyway, I kept the change in a separate commit because I wasn't sure about it

gunnarmorling replied:
The idea of the BOM is to give a user matching versions of all relevant
libs. Say you have an app using ORM@RDBMS + OGM and want to use Envers with
the former. Without that entry you'd have to match the Envers version with
the other ORM components defined by the BOM.

DavideD replied:
Ok, I'll keep it in the BOM

gunnarmorling commented on distribution/pom.xml:
```diff
@@ -37,10 +37,97 @@
+            <groupId>org.hibernate.ogm</groupId>
+            <artifactId>hibernate-ogm-couchdb</artifactId>
+        </dependency>
+        <dependency>
```
Why is Envers needed?

DavideD replied:
I don't think it's needed I'll remove it

gunnarmorling commented on distribution/pom.xml:
```diff
@@ -37,10 +37,97 @@
+            <artifactId>hibernate-envers</artifactId>
+        </dependency>
+
+        <dependency>
```
Out of interest, why are some of OGM's dependency listed here explicitly, but not all?

DavideD replied:
It's related to the way the assembly plugin works when `useTransitiveFiltering` is set to true.

Those are the dependencies that are optional or shared among different backends,
I need them as direct dependencies of the project to include them in separates folder using the dependencySet in the assembly.

gunnarmorling replied:
Ah, I see. Probably worth a short comment.

gunnarmorling commented on distribution/src/main/assembly/dist.xml:
```diff
@@ -20,254 +20,139 @@
+            <useTransitiveFiltering>false</useTransitiveFiltering>
+            <unpack>false</unpack>
+            <includes>
+                 <include>org.jboss.logging:jboss-logging-annotations</include>
```
We don't need the annotations. They are only needed at OGM's build time (we somewhat ab-use "provided" to emulate a "compile-only" scope for this).

DavideD replied:
It was part of the dist before so I kept it, I'm fine removing it

gunnarmorling commented on distribution/src/main/assembly/dist.xml:
```diff
@@ -20,254 +20,139 @@
+            <unpack>false</unpack>
+            <includes>
+                 <include>org.jboss.logging:jboss-logging-annotations</include>
+                 <include>org.jboss.spec.javax.transaction:jboss-transaction-api_1.2_spec</include>
```
As we expect it provided by the environment, should we bundle provided stuff at all? Was it part of the dist before?

DavideD replied:
Yes, it was part of the dist before, that's why I kept it

gunnarmorling replied:
Ok, cool.

gunnarmorling commented on modules/wildfly/pom.xml:
```diff
@@ -61,11 +61,6 @@
             <optional>true</optional>
         </dependency>
         <dependency>
-            <groupId>org.hibernate</groupId>
```
We don't need that anymore? Isn't it needed as part of putting the latest ORM version in place?

DavideD replied:
I dont know, none of our projects requires hibernate-infinispan and we never include it in modules or dist.

gunnarmorling replied:
+1 for not adding it to the dist.

But in case of the app server ZIPs, omitting it would leave behind the user
with a "half" ORM version, potentially falling onto their feet in case
they'd like to use ISPN as L2 cache. Not sure whether anyone is doing it,
but I'd leave this as is.

DavideD replied:
This is only the declaration of the dependency, we are not using it in the assembly that creates the module for WIldFly.
WildFly already contains hibernate-infinispan in the ORM module.

The only issue I can see is that WildFly 8.2 contains ORM 4.3.7 instead of 4.3.8 but these does not seems to cause problems at the moment.

This does not seem to cause issues, I would wait until they upgrade the dependencies in WIldFly

gunnarmorling commented:
@DavideD, Looks very good overall, added some detail remarks. Seems like it's even cleaner than before as all the transitive deps don't need to be listed anymore.

gunnarmorling commented:
The JPA API is under dist/lib/optional. Shouldn't it be under core?

DavideD commented:
@gunnarmorling 
Hi, I've applied your remarks and squashed the commit:
- I've added back hibernate-infinispan in the pom
- I've moved jboss-logging and hibernate-jpa-2.1-api in core
